drivers/potentiometer/ad5293/ad5293.c

```diff
@@ -0,0 +1,531 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include "ad5293.h"
+#include "no_os_alloc.h"
+#include "no_os_delay.h"
+
+/***************************************************************************//**
+ * @brief Reset ad5293 chip info
+ *
+ * Reset ad5293 chip info to default situation after hard/soft reset
+ *
+ * @param dev - The device structure.
+ *
+ * @return NAN
+ *******************************************************************************/
+static void ad5293_reset_chip_info(struct ad5293_dev* dev)
+{
+	uint8_t i;
+
+	for (i = 0; i < dev->chip_num; i++) {
+		(dev->chip + i)->Wiper_value = 512;
+		(dev->chip + i)->cali = CALI_PERFORMANCE;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Initialize the ad5293 device structure.
+ *
+ * Performs memory allocation of the device structure.
+ *
+ * @param device     - Pointer to location of device structure to write.
+ * @param init_param - Pointer to configuration of the driver.
+ *
+ * @return ret - return code.
+ *         Example: -ENOMEM - Memory allocation error.
+ *                  -EINVAL - No GPIO instance.
+ *                  -EIO - SPI communication error.
+ *                  -EIO - GPIO initialization error.
+ *                  0 - No errors encountered.
+ *******************************************************************************/
+int32_t ad5293_init(struct ad5293_dev **device,
+		    struct ad5293_init_param* init_param)
+{
+	struct ad5293_dev* dev;
+	struct ad5293_chip_info* chp;
+	int32_t ret = 0;
+
+	if ((! device) || (! init_param))
+		return - EINVAL;
+
+	dev = (struct ad5293_dev*)no_os_malloc(sizeof(* dev));
+	if (! dev)
+		return - ENOMEM;
+
+	printf("Initializing ad5293, number of chips %d.\r\n", init_param->chip_num);
+
+	// hardware reset pin initialization
+	if (init_param->gpio_reset) {
+		ret |= no_os_gpio_get_optional(& dev->gpio_reset, init_param->gpio_reset);
+		ret |= no_os_gpio_set_value(dev->gpio_reset, NO_OS_GPIO_HIGH);
+		if (ret < 0) {
+			ret = - EIO;
+			goto error;
+		}
+	}
+
+	// chip individual
+	dev->chip_num = init_param->chip_num;
+	chp = (struct ad5293_chip_info*)no_os_malloc(dev->chip_num * sizeof(* chp));
+	if (! chp) {
+		ret = - ENOMEM;
+		goto error;
```

Comment:
if you jump from here to `error` label, execution will attempt to free the unallocated `chp` pointer which is undefined behavior

Reply:
will add explicity initilazation to pointer dev and chp as NULL, then it won't affect even if chp is not allocated a address

Reply:
that works too